src/SmartTransactionsController.ts

```diff
@@ -330,35 +397,54 @@ export default class SmartTransactionsController extends BaseController<
     });
   }
 
-  async updateSmartTransactions() {
+  async updateSmartTransactions({
+    networkClientId,
+  }: {
+    networkClientId?: NetworkClientId;
+  } = {}): Promise<void> {
     const { smartTransactions } = this.state.smartTransactionsState;
-    const { chainId } = this.config;
+    const chainId = this.getChainId({ networkClientId });
+    const smartTransactionsForChainId = smartTransactions?.[chainId];
 
-    const currentSmartTransactions = smartTransactions?.[chainId];
-
-    const transactionsToUpdate: string[] = currentSmartTransactions
+    const transactionsToUpdate: string[] = smartTransactionsForChainId
       .filter(isSmartTransactionPending)
       .map((smartTransaction) => smartTransaction.uuid);
 
     if (transactionsToUpdate.length > 0) {
-      this.fetchSmartTransactionsStatus(transactionsToUpdate);
+      this.fetchSmartTransactionsStatus(transactionsToUpdate, {
+        networkClientId,
+      });
     }
   }
 
-  async confirmSmartTransaction(smartTransaction: SmartTransaction) {
+  // TODO make this a private method?
+  async confirmSmartTransaction(
+    smartTransaction: SmartTransaction,
+    {
+      chainId,
+      ethQuery,
+    }: {
+      chainId: Hex;
+      ethQuery: EthQuery;
+    },
+  ) {
     const txHash = smartTransaction.statusMetadata?.minedHash;
     try {
-      const transactionReceipt =
-        await this.ethersProvider.getTransactionReceipt(txHash);
-      const transaction = await this.ethersProvider.getTransaction(txHash);
+      const transactionReceipt = await query(
+        ethQuery,
+        'getTransactionReceipt',
+        [txHash],
+      );
+      const transaction = await query(ethQuery, 'getTransaction', [txHash]);
+
       const maxFeePerGas = transaction.maxFeePerGas?.toHexString();
       const maxPriorityFeePerGas =
         transaction.maxPriorityFeePerGas?.toHexString();
       if (transactionReceipt?.blockNumber) {
-        const blockData = await this.ethersProvider.getBlock(
+        const blockData = await query(ethQuery, 'getBlock', [
           transactionReceipt?.blockNumber,
           false,
-        );
+        ]);
         const baseFeePerGas = blockData?.baseFeePerGas.toHexString();
```

Comment:
This also won't work anymore since this is just a string (also `baseFeePerGas` may not be there if the network doesn't support EIP-1559). However, this is already a hex string anyway so we can just take it off:

``` suggestion
        const baseFeePerGas = blockData?.baseFeePerGas;
```

Reply:
Done here: https://github.com/MetaMask/smart-transactions-controller/pull/210/commits/94e2fac1cfdfedacd31b038a72815abb013dc19c